superset/assets/src/dashboard/components/PropertiesModal.jsx

```diff
@@ -0,0 +1,282 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Row, Col, Button, Modal, FormControl } from 'react-bootstrap';
+import Dialog from 'react-bootstrap-dialog';
+import Select from 'react-select';
+import { t } from '@superset-ui/translation';
+import { SupersetClient } from '@superset-ui/connection';
+
+import getClientErrorObject from '../../utils/getClientErrorObject';
+import withToasts from '../../messageToasts/enhancers/withToasts';
+
+const propTypes = {
+  dashboardTitle: PropTypes.string,
+  dashboardInfo: PropTypes.object,
+  owners: PropTypes.arrayOf(PropTypes.object),
+  show: PropTypes.bool.isRequired,
+  onHide: PropTypes.func,
+  onDashboardSave: PropTypes.func,
+  addSuccessToast: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  dashboardInfo: {},
+  dashboardTitle: '[dashboard name]',
+  owners: [],
+  onHide: () => {},
+  onDashboardSave: () => {},
+  show: false,
+};
+
+class PropertiesModal extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      errors: [],
+      values: {
+        dashboard_title: this.props.dashboardTitle,
+        slug: this.props.dashboardInfo.slug,
+        owners: this.props.owners || [],
+        json_metadata: JSON.stringify(this.props.dashboardInfo.metadata),
+      },
+      isOwnersLoaded: false,
+      userOptions: null,
+      isAdvancedOpen: false,
+    };
+    this.onChange = this.onChange.bind(this);
+    this.onOwnersChange = this.onOwnersChange.bind(this);
+    this.save = this.save.bind(this);
+    this.toggleAdvanced = this.toggleAdvanced.bind(this);
+  }
+
+  componentDidMount() {
+    SupersetClient.get({ endpoint: `/users/api/read` }).then(response => {
+      const options = response.json.result.map((user, i) => ({
+        // ids are in a separate `pks` array in the results... need api v2
+        value: response.json.pks[i],
+        label: `${user.first_name} ${user.last_name}`,
```

Comment:
Talking about the new endpoint, I thought the norm was `value`, `text` for select components. Is `value`, `label` what is always expected on this components? if yes I'll change it on the API, it's cleaner

Reply:
The [docs](https://v1.react-select.com/) examples use `value`, `label`.